tests/components/nextcloud/test_coordinator.py

```diff
@@ -0,0 +1,84 @@
+"""Tests for the Nextcloud coordinator."""
+
+from unittest.mock import Mock, patch
+
+from freezegun.api import FrozenDateTimeFactory
+from nextcloudmonitor import (
+    NextcloudMonitor,
+    NextcloudMonitorAuthorizationError,
+    NextcloudMonitorConnectionError,
+    NextcloudMonitorRequestError,
+)
+
+from homeassistant.components.nextcloud.const import DEFAULT_SCAN_INTERVAL
+from homeassistant.components.nextcloud.coordinator import (
+    NextcloudDataUpdateCoordinator,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from . import mock_config_entry
+from .const import NC_DATA, VALID_CONFIG
+
+from tests.common import async_fire_time_changed
+
+
+async def test_data_update(
+    hass: HomeAssistant,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test a coordinator data updates."""
+    entry = mock_config_entry(VALID_CONFIG)
+    entry.add_to_hass(hass)
+
+    with (
+        patch(
+            "homeassistant.components.nextcloud.NextcloudMonitor", spec=NextcloudMonitor
+        ) as mock_nextcloud_monitor,
+    ):
+        mock_nextcloud_monitor.return_value.update = Mock(
+            return_value=True,
+            side_effect=[
```

Comment:
This seems like this should be a parameterized test for the 3 exceptions (plus a before and after). Does that work?

Reply:
This sounds reasonable, have reworked the test to be parametrized